### PR TITLE
add adam to maintainers and codeowners

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Add Adam as a maintainer. This was +1'd by Josh, Kavitha, Peter, and Lior

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # This should match the owning team set up in https://github.com/orgs/opensearch-project/teams
-*   @pjfitzgibbons @ps48 @kavithacm @derek-ho @joshuali925 @dai-chen @YANG-DB @rupal-bq @mengweieric @vamsi-amazon @swiddis @penghuo @seankao-az @anirudha
\ No newline at end of file
+*   @pjfitzgibbons @ps48 @kavithacm @derek-ho @joshuali925 @dai-chen @YANG-DB @rupal-bq @mengweieric @vamsi-amazon @swiddis @penghuo @seankao-az @anirudha @TackAdam
```
Comment (rupal-bq) [inline .github/CODEOWNERS:2]: CODEOWNERS file showing error 
`Unknown owner on line 2: make sure @TackAdam exists and has write access to the repository`

**Review by rupal-bq — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (derek-ho) [reply .github/CODEOWNERS:2]: Yes we need to get EE on call to add him and then we should be good

**Review by kavithacm — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (gaiksaya) [reply .github/CODEOWNERS:2]: Permissions are updated!

Comment (derek-ho) [reply .github/CODEOWNERS:2]: Thanks @gaiksaya !

**Pull request merged**